File: SQLiteRepository/SQLiteQuery/DeleteQuery.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;

namespace SQLiteRepository.SQLiteQuery
{
    public class DeleteQuery : ISqliteQuery 
    {
        public string TableName { get; set; }
        public string Path { get; set; }
        public string Where { get; set; }
        public Tuple<string, List<string>> WhereIn { get; set; }
        public IQueryResult Execute()
        {
            return DeleteSqlite.Delete(this);
        }
        public IQueryResult Execute(SQLiteConnection conn, SQLiteTransaction tr)
        {
            return DeleteSqlite.Delete(conn, tr, this);
        }
    }
}

File: SQLiteAdapters/DeleteQuery.vb
Imports System.Data.SQLite

Public Class DeleteQuery
    Inherits BaseQuery
   
    Public Overrides Function Execute() As IQueryResult
        Return DeleteSqlite.Delete(Me)
    End Function
    Public Overrides Function Execute(conn As SQLiteConnection, tr As SQLiteTransaction) As IQueryResult
        Return DeleteSqlite.Delete(conn, tr, Me)
    End Function

    Public Property Delete() As List(Of String)

End Class